File: backend/src/main/java/com/military/doc/modules/document/entity/DocChangeImpact.java
package com.military.doc.modules.document.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("doc_change_impact")
public class DocChangeImpact {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long changeEventId;
    private Long impactedDocFileId;
    private String impactReason;
    private String suggestAction;
    private Long responsibleUserId;
    private String status;
    private LocalDateTime closedAt;

    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/doc-change-impact.ts
import api from './index'

export interface DocChangeImpactItem {
  id?: number
  changeEventId: number
  impactedDocFileId: number
  impactReason: string
  suggestAction: string
  responsibleUserId: number
  status: string
  closedAt: string
}

export function getDocChangeImpacts(changeEventId?: number) {
  return api.get('/doc-change-impacts', { params: { changeEventId } })
}

export function createDocChangeImpact(data: DocChangeImpactItem) {
  return api.post('/doc-change-impacts', data)
}

export function updateDocChangeImpact(id: number, data: DocChangeImpactItem) {
  return api.put(`/doc-change-impacts/${id}`, data)
}

export function deleteDocChangeImpact(id: number) {
  return api.delete(`/doc-change-impacts/${id}`)
}
